include/ddc/periodic_sampling.hpp

```diff
@@ -302,14 +316,14 @@ template <class DDim, std::enable_if_t<is_periodic_sampling_v<DDim>, int> = 0>
 KOKKOS_FUNCTION Coordinate<typename DDim::continuous_dimension_type> rmin(
         DiscreteDomain<DDim> const& d)
 {
-    return coordinate(d.front());
+    return coordinate_without_periodicity(d.front());
 }
 
 template <class DDim, std::enable_if_t<is_periodic_sampling_v<DDim>, int> = 0>
 KOKKOS_FUNCTION Coordinate<typename DDim::continuous_dimension_type> rmax(
         DiscreteDomain<DDim> const& d)
 {
-    return coordinate(d.back());
+    return coordinate_without_periodicity(d.back());
```

Comment:
Otherwise it can be negative, and `rlength()` (used in FULL normalization) returns wrong result.